src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -4890,25 +4884,16 @@ internal unsafe uint HashCaseInsensitive()
 
         public override string ToString()
         {
-            unsafe
+            int byteLength = m_StringHeapByteLength;
+            if (byteLength == 0)
             {
-                byte* buf = stackalloc byte[m_StringHeapByteLength];
-                byte* pItr = (byte*)m_pStringHeap;
-
-                for (int currentPos = 0; currentPos < m_StringHeapByteLength; currentPos++)
-                {
-                    buf[currentPos] = *pItr;
-                    pItr++;
-                }
+                return "";
+            }
 
-                if (m_StringHeapByteLength == 0)
-                    return "";
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(m_pStringHeap, byteLength);
```

Comment:
Can this just call `Encoding.GetString(ReadOnlySpan<byte> bytes)` ?

Reply:
😆 Yes :)